misc/metrics/src/protocol_stack.rs

```diff
@@ -0,0 +1,89 @@
+use libp2p_core::multiaddr::{Multiaddr, Protocol};
+use prometheus_client::encoding::text::Encode;
+
+#[derive(Encode, Hash, Clone, Eq, PartialEq)]
+pub struct Labels {
+    protocols: String,
+}
+impl Labels {
```

Comment:
I am surprised the formatter doesn't put a newline here?

Reply:
Double checked and the formatter doesn't. :shrug: 

Not sure these two blocks are distinct & separate enough to conceptually justify a blank line, but if that's how folks are used to reading it I'll do the consistent thing so it doesn't stand out (as it does not deserve to stand out).